sherpa/astro/data.py

```diff
@@ -3563,15 +3563,19 @@ def notice(self, lo=None, hi=None, ignore=False, bkg_id=None):
             return
 
         if self.units == 'energy':
-            umin = elo[0]
-            umax = ehi[-1]
+            # It's not guaranteed that channels are in increasing energy
+            # but we can certainly assume that they are monotonic.
+            # Thus, the min of the first and last entry is the minimum
+            # energy of the elo array
+            umin = min(elo[0], elo[-1])
+            umax = max(ehi[0], ehi[-1])
         elif self.units == 'wavelength':
-            umin = self._hc / ehi[-1]
-            umax = self._hc / elo[0]
+            umin = min(self._hc / ehi[[0, -1]])
+            umax = max(self._hc / elo[[0, -1]])
         else:
             # assume channel units
-            umin = self.channel[0]
-            umax = self.channel[-1]
+            umin = min(self.channel[0], self.channel[-1])
+            umax = max(self.channel[1], self.channel[-1])
```

Comment:
If we do this -ie allow channel to be in decreasing order - then shouldn't the first argument to `max` be `self.channel[0]`?

Reply:
Yes, it should, and do think we have to allow this. We won't convince XMM to change their file format and reprocess all data just because it's more convenient for us.

Reply:
I don't understand your comment here: `self.channel` is in increasing order for XMM PHA data, so why are you suggesting we change their file format. In other words, why is `umin = self.channel[0]; umax = self.chanel[-1]` not going to work? The fact that E_MIN/MAX is in decreasing order in the XMM means we have to fix our code in other places, but not in these two lines of code I think.

Reply:
Yes, this will work. 

(Forget about the other comment. I was just thinking out loud. *If* we could make XMM work with increasing energy, then this PR would not be needed. Bot that won't happen, so this PR IS needed.) I'm not sure we have to fix any other places. I work on XMM very often and I do open issues or PRs when I run into something like here. It used to work and it will work after this and the plotting PR again. But if you are aware of other areas, please do flag them.

Reply:
So, to the specific question: Of course, `umin = self.channel[0]; umax = self.chanel[-1]` works, but then the assert statement in the next line fails the assert: https://github.com/sherpa/sherpa/pull/1163/files#diff-39d64572ec8588c14cccff43dc2d85801af7524041351c8434f80963e847204bR3580

Reply:
In https://github.com/sherpa/sherpa/pull/1163#discussion_r643258528 you say that channel is in increasing order, so I don't see how `assert umin < umax` could fail.

What does

```
% dmlist xmmrgs/P0112880201R1S004SRSPEC1003.FTZ"[cols channel,counts][#row=1:10]" data,clean
```

return?

Reply:
```
(ciao) [Hanss-MacBook-Air:sherpa/sherpa-test-data/sherpatest] hamogu% dmlist xmmrgs/P0112880201R1S004SRSPEC1003.FTZ"[cols channel,counts][#row=1:10]" data,clean
#  CHANNEL COUNTS
      1          0
      2          0
      3          0
      4          0
      5          0
      6          0
      7          0
      8          0
      9          0
     10          0
```
(The first few channels are always empty...) Why exactly are you asking for this range?

Reply:
Basically it's just to check that channel is indeed increasing. So I really do not see how the `assert` fails - **unless** it's failing because it's not being triggered by the `umin = self.channel[0]/umax = self.channel[-1]` lines but why one of the other conditions.

Reply:
Ahh, Ok, now I understand. I will revert this change. It should only be needed for the energy and wavelength case.

Reply:
@DougBurke I apologize. This thread has gotten too long and I had taken this as a general discussion, while I now realize that you asked specifically about those tow lines.

Reply:
For that I think I'll assign another PR to you .....

Reply:
As if you had not done that otherwise... but I like how this is going. I feel we're making more progress in Sherpa than we did in a while.